lib/Twig/Loader/Filesystem.php

```diff
@@ -23,13 +23,17 @@ class Twig_Loader_Filesystem implements Twig_LoaderInterface, Twig_ExistsLoaderI
     protected $cache = array();
     protected $errorCache = array();
 
+    private $rootPath;
+
     /**
      * Constructor.
      *
-     * @param string|array $paths A path or an array of paths where to look for templates
+     * @param string|array $paths    A path or an array of paths where to look for templates
+     * @param string       $rootPath The root path common to all relative paths
      */
-    public function __construct($paths = array())
+    public function __construct($paths = array(), $rootPath = null)
     {
+        $this->rootPath = realpath(null !== $rootPath ? $rootPath : getcwd()).DIRECTORY_SEPARATOR;
```

Comment:
well, if someone uses relative path for the loader paths and passes ``__DIR__`` as root path (instead of making all paths absolute in the loader paths), your suggestion would break the project once it gets bundled as phar.
This seems worse than forbidding passing relative paths to the loader altogether.